rz-cmn:  fix kernel panic yolovx test app

long-trinh-rvc opened the pull request.

khanhduydo-rvc committed:
rz-cmn: Update mmngr driver following kernel 6.1

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

long-trinh-rvc committed:
rz-cmn: kernel-module-mmngr: add patches for lossy mem and OMXBUF size

Changes include:
    - Disable parsing/initialization of lossy mem area for RZ/G2L, RZ/G3E, and RZ/V2H series
    - Define MM_OMXBUF_SIZE for RZ/G2L, RZ/G3E, and RZ/V2H

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

long-trinh-rvc committed:
rz-cmn: kernel-module-mmngr: add dynamic MM_OMXBUF_SIZE patch

Previously, MM_OMXBUF_SIZE was changed at build stage. Now all supported
devices are combined into a single build with MMNGR_SALVATORX defining a
fixed size.

To properly handle this and avoid future hot‑fixes, the driver is updated
to determine the buffer size at run time for supported boards, while
keeping the legacy workflow intact. MM_OMXBUF_SIZE can still be set at
build time first, and is only overridden if the board is in the supported
list.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

vudangRVC commented:
Hi @long-trinh-rvc , is there any reason to add dynamic size updates?
Please mention any reason in the patch.

long-trinh-rvc commented:
Hi @vudangRVC ,

Back then, MM_OMXBUF_SIZE was changed dynamically at build stage. Now we combine all supported devices into a single build and define MMNGR_SALVATORX to set a fixed MM_OMXBUF_SIZE (see [reference](https://github.com/Renesas-SST/meta-renesas/blob/styhead/rz-cmn/recipes-kernel/kernel-module-mmngr/kernel-module-mmngr.bb#L27)).

To properly handle this dynamic size—and avoid future hot‑fixes like [this one](https://github.com/Renesas-SST/meta-renesas/blob/dunfell/rz-sbc/meta-rz-common/recipes-kernel/kernel-module-mmngr/kernel-module-mmngr/0001-Reduce-MM_OMXBUF_SIZE-for-omx.patch)—I updated the driver to determine the buffer size at run time for supported boards, while keeping the legacy workflow intact. MM_OMXBUF_SIZE can still be set and changed at build time first, and the size is only overridden if the board is in the supported list.

This also addresses [Cong Nguyen’s concern](https://jira.eng.renesas.com/browse/AMECSSTSWA-312#:~:text=FYI%2C%20The%20MM_OMXBUF_SIZE,MM_OMXBUF_SIZE%20for%20V2H.) that V2H and G2L require different MM_OMXBUF_SIZE values, ensuring correct handling across boards.

vudangRVC commented:
Then, please make sure this patch: https://github.com/Renesas-SST/meta-renesas/pull/112/files#diff-37b1ad7777c75b58157fed44a5b157978ef5275d7bb8f448eeaff3ebe9e0a099 has a detailed message

long-trinh-rvc commented:
I have updated

long-trinh-rvc committed:
rz-cmn: linux-yocto: remove redundant overlay from recipe

Follow-up to https://github.com/Renesas-SST/meta-renesas/pull/111/changes.
Removed the redundant overlay entry from the linux recipe to fix build
failure:

| make[4]: *** No rule to make target 'arch/arm64/boot/dts/renesas/overlays/rzv2h-rdk-1.0-audio-hdmi.dtbo'. Stop.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

vudangRVC commented on recipes-kernel/kernel-module-mmngr/kernel-module-mmngr.bb:
```diff
@@ -17,11 +17,15 @@ SRC_URI:append = " \
     file://0003-mmngr_drv-mmngr-module-drv-Update-physical-convert-f.patch \
     file://0004-mmngr-lock-mm-read-before-finding-VMA.patch \
     file://0005-rzg2l-sbc-mmngr-supports-linux-6.10.patch \
+    file://0006-Handle-MM_OMXBUF_SIZE-dynamically.patch \
```
The order is inconsistent. Can you update it to sequential numbering?

long-trinh-rvc replied:
I've updated

tiennguyenuh commented on recipes-kernel/kernel-module-mmngr/kernel-module-mmngr/0014-rz-cmn-mmngr-supports-linux-6.10.patch:
@long-trinh-rvc , what is the purpose of this patch? Could you please add more details to the commit message?

tiennguyenuh replied:
Also, the patch name is specific to the rzg2l-sbc. Is that intentional?

long-trinh-rvc replied:
That patch was already in place before. I just renamed it

tiennguyenuh replied:
If the patch isn't good, why not improve it

long-trinh-rvc replied:
The main purpose of that patch is to update to 6.10 API:
[Refer commit](https://github.com/Renesas-SST/linux-rz/commit/1c71222e5f2393b5ea1a41795c67589eea7e3490)
I will rename it to ```0014-rz-cmn-mmngr-supports-linux-6.10.patch```

tiennguyenuh commented on recipes-kernel/kernel-module-mmngr/kernel-module-mmngr/0015-Handle-MM_OMXBUF_SIZE-dynamically.patch:
```diff
@@ -0,0 +1,91 @@
+-	{ .family = "RZ/G2UL" },
+-	{ .family = "RZ/G3E" },
+-	{ .family = "RZ/V2H" },
++	{ .family = "RZ/G2L", .data = (void *)(128 * 1024 * 1024UL) },
++	{ .family = "RZ/G2UL", .data = (void *)(128 * 1024 * 1024UL) },
++	{ .family = "RZ/G3E", .data = (void *)(256 * 1024 * 1024UL) },
++	{ .family = "RZ/V2H", .data = (void *)(256 * 1024 * 1024UL) },
```
I think .data is already defined in `mmngr/include/mmngr_private.h`. You can reuse it instead of hardcoding it here.

long-trinh-rvc replied:
The data in ```mmngr/include/mmngr_private.h``` are for build time. I leave them unchanged, just in case anyone wants to use that method to build Yocto. For supported boards, mmngrbuf will use the overridden matching .data to allocate the exact memory for that board.
